SEARCHSPEC.md

```diff
@@ -0,0 +1,1476 @@
+        *   `extract(field FROM x) → bigint*`
+        *   `from_unixtime(unixtime) -> timestamp(3)*`
+        *   `from_unixtime(unixtime, zone) → timestamp(3) with time zone*`
+        *   `from_unixtime(unixtime, hours, minutes) → timestamp(3) with time zone*`
+        *   `Localtime*`
+        *   `localtimestamp*`
+        *   `localtimestamp(p)*`
+        *   `now() → timestamp(3) with time zone*`
+        *   `to_unixtime(timestamp) → double*`
+    *   **MySQL-like date functions:**
+        *   `date_format(timestamp, format) → varchar*`
+        *   `date_parse(string, format) -> timestamp(3)*`
+*   **Aggregate functions \
+**Note that Presto provides a much larger superset of functions. Bitwise, map, and approximate aggregations are mostly absent. Only BigQuery has a few native approximate aggregation functions.
+    *   `array_agg(x) → array&lt;[same as input]>*`
+    *   `avg(x) → double`
+    *   `bool_and(boolean) → boolean*`
+    *   `bool_or(boolean) → boolean*`
+    *   `count(*) → bigint*`
+    *   `count(x) → bigint`
+    *   `count_if(x) → bigint*`
+    *   `every(boolean) → boolean*`
+    *   `max(x) → [same as input]`
+    *   `max(x, n) → array&lt;[same as x]>*`
+    *   `min(x) → [same as input]`
+    *   `min(x, n) → array&lt;[same as x]>*`
+    *   `sum(x) → [same as input]`
+    *   **Statistical Aggregate Functions:**
+        *   `corr(y, x) → double*`
+        *   `covar_pop(y, x) → double*`
+        *   `covar_samp(y, x) → double*`
+        *   `stddev(x) → double`
+        *   `stddev_pop(x) → double`
+        *   `stddev_samp(x) → double`
+        *   `variance(x) → double`
+        *   `var_pop(x) → double`
+        *   `var_samp(x) → double`
+*   **Window functions**
+    *   **Ranking Functions:**
+        *   `cume_dist() → bigint`
+        *   `dense_rank() → bigint`
+        *   `ntile(n) → bigint`
+        *   `percent_rank() → double`
+        *   `rank() → bigint`
+        *   `row_number() → bigint`
+    *   **Value Functions:**
+        *   `first_value(x) → [same as input]`
+        *   `last_value(x) → [same as input]`
+        *   `nth_value(x, offset) → [same as input]`
+        *   `lead(x[, offset[, default_value]]) → [same as input]`
+        *   `lag(x[, offset[, default_value]]) → [same as input]`
+*   **JSON functions 
+**In general, function signatures and behaviour differs across implementations for many JSON related functions.
+    *   `json_array_length(json) → bigint*`
+    *   `json_extract(json, json_path) → json*`
+    *   `json_extract_scalar(json, json_path) → varchar*`
+    *   `json_format(json) → varchar*`
+    *   `json_size(json, json_path) → bigint*`
+*   Functions for working with nested and repeated data (ROW and ARRAY) \
+See also UNNEST, which is part of the SQL grammar and allows working with nested arrays as if they were rows in a joined table. \
+Note: Arrays are mostly absent in MySQL
+    *   Array subscript Operator: []
```

Comment:
Formatting is broken here, possibly because of the lines above that end with `\`